eventauth/eventcontent.go

```diff
@@ -0,0 +1,280 @@
+package eventauth
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+)
+
+// creaateContent is the JSON content of a m.room.create event along with
+// the top level keys needed for auth.
+type createContent struct {
+	// We need the domain of the create event when checking federatebility.
+	senderDomain string
+	// We need the roomID to check that events are in the same room as the create event.
+	roomID string
```

Comment:
Hmm, it's interesting. The auth code needs both top-level keys from the create event object, and the parsed create event content.
It turned out easier to write that code by shuffling the top-level keys into the same object as the create event content, rather than passing the two around separately.